platform.md

```diff
@@ -1376,3 +1386,36 @@ Where:
 }
 ```
 This label MUST contain the JSON representation of [`project-metadata.toml`](#project-metadatatoml-toml)
+
+
+## Deprecations
+This section describes all the features that are deprecated.
+
+### `io.buildpacks.stack.*` Labels
+
+_Deprecated in Platform API 0.12._
+
+For compatibility with older platforms and older buildpacks, base image authors SHOULD ensure for build images and run images:
+
+- The image config's `Env` field has the environment variable `CNB_STACK_ID` set to the stack ID.
+- The image config's `Label` field has the label `io.buildpacks.stack.id` set to the stack ID.
+- The image config's `Label` field has the label `io.buildpacks.stack.mixins` set to a JSON array containing mixin names for each mixin applied to the image.
+
+To upgrade, the platform SHOULD upgrade all buildpacks to use Buildpack API `0.10` or greater.
+
+### `io.buildpacks.lifecycle.metadata` (JSON) `stack` Key
```

Comment:
i guess to my question from above,  does it make sense to move most of the deleted stuff about stacks down into this section and basically say like, 

> the lifecycle SHOULD prefer the spec above but MUST fall back on the deprecated stack logic until [ the heat death of the universe | buildpacks 1.0 | end of calendar year 2024 | we say so] ?

Reply:
Yeah... although, I would prefer to only add the stuff that falls within the responsibility of the lifecycle. So stuff like:
> The image config's `Label` field has the label `io.buildpacks.stack.maintainer` set to the name of the stack maintainer.

I would prefer to leave off because it really makes no difference to the lifecycle if that label is set or not. Mixins are kind of in a similar boat, even though somewhere I think we say that the lifecycle validates mixins, it does not. That responsibility is left up to the platform. If a platform wants to support earlier platform API versions, it can consult the `platform/0.X` version of the spec.

Other two points might be worth moving or repeating here:
> **If** `<run-image>` is not provided by the platform, the value will be [resolved](#run-image-resolution) from the contents of the `runImage` key in the `io.buildpacks.lifecycle.metdata` label on `<image>`, or `stack.runImage` if not found (for compatibility with older platforms).

> | `CNB_STACK_ID` | Chosen stack ID

This variable will be passed-through by the lifecycle to buildpacks regardless of the buildpack API in use (in the current implementation).

I'm a bit mixed on moving everything into this section vs clearly marking things as deprecated where they are mentioned. We have preferred the former but it makes understanding deprecations a bit confusing.